# --- Configurazione dell'Ambiente di Simulazione ---

# Descrizione della mappa a griglia.
# Il simulatore caricherà questa immagine e la userà per il ray-casting del laser.
map:
  # Percorso del file immagine. I pixel neri sono ostacoli, i bianchi sono liberi.
  image_path: "$(find multi_robot_simulator)/maps/my_map.pgm"  
  # Risoluzione: quanti metri reali rappresenta un singolo pixel.
  resolution: 0.05  # metri/pixel
  # Origine: coordinate [x, y, yaw] dell'angolo in basso a sinistra dell'immagine
  # nel mondo del simulatore. Utile per allineare la mappa.
  origin: [-10.0, -10.0, 0.0]

# --- Lista e Configurazione dei Robot ---

# 'robots' è una lista. Ogni elemento che inizia con '-' è un nuovo robot.
robots:
  # --- Robot 1 ---
  - name: "robot1"
    frame_id: "robot1/base_link" # Nome del frame TF principale del robot (convenzione ROS)
    initial_pose: [0.0, 0.5, 0.0] # Posizione iniziale [x (m), y (m), theta (radianti)]
    
    # Proprietà specifiche del robot, come richiesto dalle slide
    max_velocities:
      linear_x: 0.5   # m/s
      angular_z: 1.0  # rad/s

    # Lista dei sensori montati SU QUESTO robot. Anche questa è una lista.
    sensors:
      # Sensore Laser per robot1
      - type: "laser_scanner"
        frame_id: "robot1/laser_frame" # Nome del frame TF del sensore
        topic: "/robot1/scan"          # Topic su cui pubblicare i dati
        
        # Posizione del sensore rispetto al centro del robot (frame_id del robot)
        # [x, y, theta] -> 10cm in avanti, 0cm di lato, 0 rotazione
        relative_pose: [0.1, 0.0, 0.0] 
        
        # Proprietà specifiche del Lidar, come richiesto dalle slide
        beams: 360              # Numero di raggi laser
        range_min: 0.1          # Distanza minima di rilevamento (m)
        range_max: 10.0         # Distanza massima di rilevamento (m)
        angle_min: -3.14159     # Angolo del primo raggio (radianti, -180°)
        angle_max: 3.14159      # Angolo dell'ultimo raggio (radianti, +180°)

  # --- Robot 2 ---
  - name: "robot2"
    frame_id: "robot2/base_link"
    initial_pose: [1.0, -1.0, 1.57] # Parte già ruotato di 90°
    
    max_velocities:
      linear_x: 0.4
      angular_z: 1.2

    sensors:
      # Sensore Laser per robot2 (potrebbe avere caratteristiche diverse)
      - type: "laser_scanner"
        frame_id: "robot2/laser_frame"
        topic: "/robot2/scan"
        relative_pose: [0.1, 0.0, 0.0]
        beams: 180
        range_min: 0.2
        range_max: 8.0
        angle_min: -1.5707 # Un laser a 180° frontale
        angle_max: 1.5707
